hudi-common/src/test/java/org/apache/hudi/common/table/timeline/TestHoodieInstant.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.timeline;
+
+import org.apache.hudi.common.testutils.HoodieCommonTestHarness;
+import org.apache.hudi.common.util.Option;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import static org.apache.hudi.common.testutils.Assertions.assertStreamEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestHoodieInstant extends HoodieCommonTestHarness {
+
+  @Test
+  public void testExtractTimestamp() {
+    String fileName = "20230104152218702.inflight";
+    assertEquals("20230104152218702", HoodieInstant.extractTimestamp(fileName));
+
+    fileName = "20230104152218702.commit.request";
+    assertEquals("20230104152218702", HoodieInstant.extractTimestamp(fileName));
+  }
+
+  @Test
+  public void testGetTimelineFileExtension() {
+    String fileName = "20230104152218702.inflight";
+    assertEquals(".inflight", HoodieInstant.getTimelineFileExtension(fileName));
+
+    fileName = "20230104152218702.commit.request";
+    assertEquals(".commit.request", HoodieInstant.getTimelineFileExtension(fileName));
+  }
+
+  @Test
+  public void testCreateHoodieInstantByFileStatus() throws IOException {
+    try {
+      initMetaClient();
+      HoodieInstant instantRequested =
+          new HoodieInstant(HoodieInstant.State.REQUESTED, HoodieTimeline.COMMIT_ACTION, "001");
+      HoodieInstant instantCommitted =
+          new HoodieInstant(HoodieInstant.State.COMPLETED, HoodieTimeline.COMMIT_ACTION, "001");
+      HoodieActiveTimeline timeline = metaClient.getActiveTimeline();
+      timeline.createNewInstant(instantRequested);
+      timeline.transitionRequestedToInflight(instantRequested, Option.empty());
+      timeline.saveAsComplete(
+          new HoodieInstant(true, instantRequested.getAction(), instantRequested.getTimestamp()),
+          Option.empty());
+      metaClient.reloadActiveTimeline();
```

Comment:
We also need some tests for incremental data source.

Reply:
We should not use `stateTransitionTime` as default true, or this could be a breaking change for all the existing jobs.

Reply:
changed

Reply:
Can we also add a test when the file is moved, the state transition time also changes.

Reply:
Do you mean the instant file is moved?

Reply:
Yes. The test expect to fail in such use case.

Reply:
There're different behaviors for HDFS and object stores, for HDFS, it's modification time won't change if it's a rename operation, but since many object store use `COPY + DELETE` operations to perform `move`, the modification time will be updated in such cases.